src/pages/api/orgs/[orgId]/events.ics/index.ts

```diff
@@ -0,0 +1,74 @@
+import dayjs, { Dayjs } from 'dayjs';
+import { NextApiRequest, NextApiResponse } from 'next';
+
+import BackendApiClient from 'core/api/client/BackendApiClient';
+import { ZetkinEvent } from 'utils/types/zetkin';
+
+const timeStamp = (t: Dayjs): string => t.format('YYYYMMDDTHHmmss');
+const formatString = (str: string): string =>
+  str
+    .replaceAll('\n', '\\n')
+    .match(/.{1,75}/g)
+    ?.join('\n\t') ?? '';
+
+export default async function handle(
+  req: NextApiRequest,
+  res: NextApiResponse
+): Promise<void> {
+  const { orgId } = req.query;
+
+  const apiClient = new BackendApiClient(req.headers);
+
+  const startTime = dayjs()
+    .subtract(2, 'month')
+    .toDate()
+    .toISOString()
+    .slice(0, 10);
+
+  const events = await apiClient.get<ZetkinEvent[]>(
+    `/api/orgs/${orgId}/actions?frecursive&ilter=start_time>=${startTime}`
+  );
+
+  const vEvents: string[] = [];
+  events
+    .filter((event) => !event.cancelled && event.published)
+    .forEach((event) => {
+      vEvents.push(`BEGIN:VEVENT`);
+      vEvents.push(`UID:${event.id}@zetkin.org`);
+      vEvents.push(
+        `ORGANIZER;CN=${event.organization.title.replace(
+          '\n',
+          ' '
+        )}:MAILTO:noreply@zetkin.org`
+      );
+      vEvents.push(`DTSTAMP:${timeStamp(dayjs(event.published))}`);
+      vEvents.push(`DTSTART:${timeStamp(dayjs(event.start_time))}`);
+      vEvents.push(`DTEND:${timeStamp(dayjs(event.end_time))}`);
+      vEvents.push(`SUMMARY:${event.title ?? ''}`);
```

Comment:
Is it a conscious choice to use an empty string as fallback? In other places, we use a localized message for "Untitled event" or equivalent, but maybe using an empty string makes sense here because the client is expected to have it's own fallback?

Reply:
I also like the idea that we let the client figure it out!